backends/tc/tc_defines.h

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright (C) 2023 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions
+and limitations under the License.
+*/
+
+#ifndef BACKENDS_TC_TC_DEFINES_H_
+#define BACKENDS_TC_TC_DEFINES_H_
+
+#define DEFAULT_TABLE_ENTRIES 2048
```

Comment:
@Sosutha Can we make all of this constexpr within a dedicated namespace? They are clashing with other #defines. In general, I would consider that a better idea than to declare a list of global defines.